crates/bevy_ecs/src/component.rs

```diff
@@ -193,6 +194,27 @@ impl ComponentDescriptor {
         }
     }
 
+    /// Create a new `ComponentDescriptor`.
+    ///
+    /// # Safety
+    /// - the `drop` fn must be usable on a pointer with a value of the layout `layout`
+    /// - the component type must be safe to access from any thread (Send + Sync in rust terms)
+    pub unsafe fn new_with_layout(
+        name: String,
+        storage_type: StorageType,
+        layout: Layout,
+        drop: unsafe fn(*mut u8),
+    ) -> Self {
+        Self {
+            name,
+            storage_type,
+            is_send_and_sync: true,
```

Comment:
I think this should be an argument too.

Reply:
bevy currently cannot handle non-send components, only resources. Even if this was configurable queries would happily let the user access the components on whatever thread they like.

So for now I went with requiring Send+Sync in the safety section of this function, and if there's ever a use case for non-send component we can implement them and make this is an argument.